table.go

```diff
@@ -608,8 +608,16 @@ func ValuesToBuffer(schema *dynparquet.Schema, vals ...any) (*dynparquet.Buffer,
 						pair := reflect.ValueOf(dynVals.Index(j).Interface())
 						dynamicColumns[col.Name] = append(dynamicColumns[col.Name], ToSnakeCase(fmt.Sprintf("%v", pair.Field(0))))
 					}
+				case reflect.Map:
+					dynVals := reflect.ValueOf(cv)
+					for _, key := range dynVals.MapKeys() {
+						if key.Kind() != reflect.String {
```

Comment:
Can we make this a switch statement and just have the default case return the error?

Reply:
Sure thing -- made the fix. Thanks @thorfour